pkg/apis/kops/cluster.go

```diff
@@ -452,6 +452,10 @@ type EtcdClusterSpec struct {
 	MemoryRequest *resource.Quantity `json:"memoryRequest,omitempty"`
 	// CPURequest specifies the cpu requests of each etcd container in the cluster.
 	CPURequest *resource.Quantity `json:"cpuRequest,omitempty"`
+	// MetricsPort specifies port to expose /metrics endpoint.
+	MetricsPort int `json:"metricsPort,omitempty"`
+	// MetricsType specifies level of detail for exported metrics.
+	MetricsType string `json:"metrics,omitempty"`
```

Comment:
I think this is what @rifelpet proposes. This way you can just compare it to nil to check if it's set.
```suggestion
	// MetricsPort specifies port to expose /metrics endpoint.
	MetricsPort *int `json:"metricsPort,omitempty"`
	// MetricsType specifies level of detail for exported metrics.
	MetricsType *string `json:"metrics,omitempty"`
```

Reply:
The rest of EtcdClusterSpec doesn't use pointers for primitive types, so maybe it's better to be consistent? and if we want to switch to pointers down the road we can do it all at once.

Reply:
Exactly, due to using primitive types I've decided to not use pointers here.